elastic/kibana
# [Security Solution] Integrate state and components for Prebuilt Rule Update Workflow

**Epic:** https://github.com/elastic/kibana/issues/174168
**Addresses:** https://github.com/elastic/kibana/issues/171520

## Summary

This PR introduces a new `Update` tab allowing users to resolve rule upgrade conflicts. It's a result of combination of read-only components implemented in https://github.com/elastic/kibana/pull/193261 and rule upgrade state implemented in https://github.com/elastic/kibana/pull/191721.

## Details

The goal of this PR is to provide intermediate integration between rule upgrade state ([PR](https://github.com/elastic/kibana/pull/191721)) and components displaying the diff and read-only state ([PR](https://github.com/elastic/kibana/pull/193261)). It will facilitate further development of rule field editable components and streamline rule upgrade functionality developing.

## How to test?

The functionality is hidden under `prebuiltRulesCustomizationEnabled` feature flag. Add the following to your Kibana config

```yaml
xpack.securitySolution.enableExperimental:
  - prebuiltRulesCustomizationEnabled
```

When the above feature flag enabled the new `Update` tab is displayed instead of the old one.

## Screenshots

Suggested components design 
![image](https://github.com/user-attachments/assets/b5aaf571-286a-4595-9bd4-fdaf9a423b03)

New `Update` tab
<img width="1718" alt="image" src="https://github.com/user-attachments/assets/28aa6bb3-f805-4109-a808-d67e58c7c5b8">
